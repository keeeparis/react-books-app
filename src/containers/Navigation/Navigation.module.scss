.Container {
  position: sticky;
  top: 0;
  min-height: 60px;
  background-color: var(--color-3-dark);
  color: white;
  display: flex;
  z-index: 2;
  box-shadow: 1px 6px 20px 1px rgb(0 0 0 / 14%);
}

.Wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 80px;
  font-weight: 400;
  flex: 1;

  a {
    text-decoration: none;
    color: inherit;

    &:hover {
      text-decoration: underline;
    }
  }
}

@media (max-width: 660px) {
  .Wrapper {
    padding: 0 40px;
  }
}

.Up {
  cursor: pointer;
  &:hover {
    text-decoration: underline;
  }
}
