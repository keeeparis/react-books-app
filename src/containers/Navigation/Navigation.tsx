import React, { MouseEvent } from 'react'
import { useTranslation } from 'react-i18next'
import { Link, useLocation, useNavigate } from 'react-router-dom'
import LanguagePicker from '../../components/LanguagePicker'

import styles from './Navigation.module.scss'

const Navigation = () => {
  const location = useLocation()
  const navigate = useNavigate()
  const { t } = useTranslation()

  const isIndexPage = location.pathname === '/'
  const link = isIndexPage ? (
    <Link to={'/'}>{t('home')}</Link>
  ) : (
    <a onClick={() => navigate(-1)}>{t('back')}</a>
  )

  return (
    <div className={styles.Container}>
      <div className={styles.Wrapper}>
        {link}
        <div className={styles.RightSide}>
          <LanguagePicker />
        </div>
      </div>
    </div>
  )
}

export default Navigation
