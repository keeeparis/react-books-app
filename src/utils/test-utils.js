import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
