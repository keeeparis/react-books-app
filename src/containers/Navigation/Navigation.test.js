import { render, screen, fireEvent, userEvent } from 'test-utils'
import Navigation from './Navigation'

it('test Navigation Component', () => {
  const { asFragment } = render(<Navigation />)

  expect(asFragment()).toMatchSnapshot()
})
