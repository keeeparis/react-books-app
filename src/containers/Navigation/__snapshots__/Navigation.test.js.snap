// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navigation Component 1`] = `
<DocumentFragment>
  <div
    class="Container"
  >
    <div
      class="Wrapper"
    >
      <a
        href="/"
      >
        Home
      </a>
      <div
        class="RightSide"
      >
        <div
          class="Wrapper"
        >
          <div
            class="select"
          >
            <div
              class="selectShow"
              role="selected"
              tabindex="0"
            >
              en
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
